orttraining/orttraining/eager/opgen/opgen/generator.py

```diff
@@ -328,7 +328,30 @@ def _write_function_body(self, writer: writer.SourceWriter, mapped_func: MappedO
                 )
                 writer.writeline()
 
+        last_op_index = len(ctx.ops) - 1
+        equal_onnxop_found = False
+        return_info = cpp_func.torch_func.return_type if cpp_func.torch_func else None
+        # check if the return is a ref tensor and there exists an out aten method input
+        set_out_tensor = 0
+        last_param = cpp_func.parameters[-1].member
+        if return_info and last_param and last_param.identifier.value == "out":
+            for torch_p in last_param.torch_param:
+                output_alias = self._get_alias_info(return_info)
+                if output_alias and self._get_alias_info(torch_p) == output_alias and output_alias.is_writable:
+                    set_out_tensor = 1
+                    writer.writeline("// resize the output and then create output ort value to be updated.")
+                    writer.writeline(
+                        "resize_output(invoker, dynamic_cast<ORTTensorImpl*>(out.unsafeGetTensorImpl()), self.sizes());"
+                    )
+                    writer.writeline(f"auto ort_input_out = create_ort_value(invoker, out);")
+                    writer.writeline()
+
         for onnx_op_index, onnx_op in enumerate(ctx.ops):
+            # equal onnx op only returns bool but that doesn't align with aten behavior,
+            # so we track if the Equal op is in the list and if so we'll do a cast at the end
+            if onnx_op.name == "Equal":
```

Comment:
what if equal is not the last op?

Reply:
add comment about this in atenops.py. currently it doesn't need to be last for ops like ne.